public/app/features/dashboard-scene/settings/GeneralSettings.tsx

```diff
@@ -22,11 +69,190 @@ export class GeneralSettingsEditView
   static Component = ({ model }: SceneComponentProps<GeneralSettingsEditView>) => {
     const dashboard = getDashboardSceneFor(model);
     const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+    const { title, description, tags, meta, editable, graphTooltip, overlay } = dashboard.useState();
+    const timeRange = sceneGraph.getTimeRange(dashboard);
+    const refreshPicker = getRefreshPickerControl(dashboard);
+
+    const onTitleChange = (value: string) => {
```

Comment:
It's a practice we use to implement model-modifying methods on the model directly, rather than in the component. This way the code can be easily unit tested without the need to render the component